### PR TITLE
Feature/genesisfile creator

### PR DESCRIPTION
**Pull request by n-hutton**

### CODE REVIEW

File: scripts/end_to_end_test/run_end_to_end_test.py [hunk 1]
```diff
@@ -175,6 +175,22 @@ def start_node(self, index):
 
         time.sleep(0.5)
 
+    def restart_node(self, index):
+        print('Restarting Node {}...'.format(index))
+
+        self._nodes[index].stop()
+
+        # Optically remove db files when testing recovering from a genesis file
+        if False:
```
Comment (ejfitzgerald) [inline scripts/end_to_end_test/run_end_to_end_test.py:184]: Make an option?

File: libs/ledger/src/genesis_file_creator.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  ledger::Block genesis_block;
+
+  genesis_block.body.merkle_hash  = merkle_commit_hash;
+  genesis_block.body.block_number = 0;
+  genesis_block.body.miner        = ledger::Address(crypto::Hash<crypto::SHA256>(""));
+  genesis_block.UpdateDigest();
+
+  FETCH_LOG_INFO(LOGGING_NAME, "Created genesis block hash: ", genesis_block.body.hash.ToBase64());
+
+  ledger::GENESIS_MERKLE_ROOT = merkle_commit_hash;
```
Comment (ejfitzgerald) [inline libs/ledger/src/genesis_file_creator.cpp:139]: This feels potentially raced but fine for a v1

**Review by ejfitzgerald — APPROVED** [review]

File: scripts/end_to_end_test/run_end_to_end_test.py (hunk 1 above)
Comment (n-hutton) [reply scripts/end_to_end_test/run_end_to_end_test.py:184]: it's going to be useful to keep that there for a short while as I will have to add an e2e test for the custom genesis block etc.

File: libs/ledger/src/genesis_file_creator.cpp (hunk 2 above)
Comment (n-hutton) [reply libs/ledger/src/genesis_file_creator.cpp:139]: You're right, there is one potential race in main chain rpc service here:    ``` if (it->body.previous_hash == GENESIS_DIGEST)```

Addressed in another PR. I am now fairly confident there are no other races at this time

**Pull request merged**